Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -938,8 +828,7 @@ static inline bool shouldAddContentKeyRecipients()
     if (!m_cdmInstance && !m_session.get())
         return false;
 
-    // DecompressionSessions doesn't support encrypted media.
-    if (isEnabledVideoTrackID(trackID) && !m_videoRenderer)
+    if (!isEnabledVideoTrackID(trackID))
         return false;
```

Comment:
We shouldn't check `isEnabledVideoTrackID(trackID)` twice within the same function. Consider moving the last `if (!isEnabledVideoTrackID(trackID))` check to the top of the function and making the first `if (isEnabledVideoTrackID(trackID) && !m_isSelectedForVideo)` just check for `!m_isSelectedForVideo`.